### PR TITLE
Mk 1966 add unit  tests for taquito-contracts-library

### LINKED ISSUE #1966: mutation test and extend test coverage for package: taquito-contract-library
Run stryker on taquito-contract-library package and try to eliminate mutants and extend coverage so it is reported in green on the unit test summary

### PR DESCRIPTION
**Pull request by michaelkernaghan**
bring the coverage up to the green where possible, and remove mutants where practical.

CLOSE #1966

### CODE REVIEW

File: packages/taquito-contracts-library/test/rpc-wrapper.spec.ts [hunk 1]
```diff
@@ -28,15 +28,6 @@ describe('RpcWrapperContractsLibrary tests', () => {
     done();
   });
 
-  it('get entrypoints from RPC when contract is not in the library', async (done) => {
```
Comment (roxaneletourneau) [inline packages/taquito-contracts-library/test/rpc-wrapper.spec.ts:31]: Why this test is removed?

File: packages/taquito-contracts-library/test/taquito-contracts-library.spec.ts [hunk 2]
```diff
@@ -90,19 +101,36 @@ describe('ContractsLibrary tests', () => {
           script,
           entrypoints,
         },
+    
       });
     } catch (e: any) {
+      expect
```
Comment (roxaneletourneau) [inline packages/taquito-contracts-library/test/taquito-contracts-library.spec.ts:107]: ~expect~

File: packages/taquito-contracts-library/test/taquito-contracts-library.spec.ts [hunk 3]
```diff
@@ -90,19 +101,36 @@ describe('ContractsLibrary tests', () => {
...
+      expect(validateAddress(contractAddress)).toEqual(0)
```
Comment (roxaneletourneau) [inline packages/taquito-contracts-library/test/taquito-contracts-library.spec.ts:108]: validateAddress is a function from the '@taquito/utils'. I don't think it should be tested here.

File: packages/taquito-contracts-library/test/taquito-contracts-library.spec.ts [hunk 4]
```diff
@@ -90,19 +101,36 @@ describe('ContractsLibrary tests', () => {
...
       expect(e).toBeInstanceOf(InvalidAddressError);
       expect(e.message).toEqual(`Address is invalid: ${contractAddress}`);
       expect(e).toBeInstanceOf(Error);
     }
   });
 
+  it('do not throw an InvalidAddress error if the contract address is valid', () => {
```
Comment (roxaneletourneau) [inline packages/taquito-contracts-library/test/taquito-contracts-library.spec.ts:115]: Is it intended that there is no expect statement in the test?

File: packages/taquito-contracts-library/test/taquito-contracts-library.spec.ts [hunk 5]
```diff
@@ -90,19 +101,36 @@ describe('ContractsLibrary tests', () => {
...
+      },
+    });
+  });
+
   it('throw an InvalidScriptFormatError error if the script format is invalid', () => {
     const contractAddress = 'KT1NGV6nvvedwwjMjCsWY6Vfm6p1q5sMMLDY';
     const contractLib = new ContractsLibrary();
     const script: any = 'invalid';
 
+    !expect(script.code);
```
Comment (roxaneletourneau) [inline packages/taquito-contracts-library/test/taquito-contracts-library.spec.ts:132]: First time I see except used that way. Can you explain what it does?

File: packages/taquito-contracts-library/test/taquito-contracts-library.spec.ts [hunk 6]
```diff
@@ -118,4 +146,20 @@ describe('ContractsLibrary tests', () => {
       expect(e).toBeInstanceOf(Error);
     }
   });
-});
+
+  it('Test the configure context function through use of Tezos.addExtension', () => {
```
Comment (roxaneletourneau) [inline packages/taquito-contracts-library/test/taquito-contracts-library.spec.ts:150]: Unit tests should test specific functions. Testing Tezos.addExtension might be more appropriate in the taquito package

**Review by roxaneletourneau — CHANGES_REQUESTED** [review]

File: packages/taquito-contracts-library/test/rpc-wrapper.spec.ts (hunk 1 above)
Comment (michaelkernaghan) [reply packages/taquito-contracts-library/test/rpc-wrapper.spec.ts:31]: A mistake. I have restored the test.

File: packages/taquito-contracts-library/test/taquito-contracts-library.spec.ts (hunk 2 above)
Comment (michaelkernaghan) [reply packages/taquito-contracts-library/test/taquito-contracts-library.spec.ts:107]: removed it

File: packages/taquito-contracts-library/test/taquito-contracts-library.spec.ts (hunk 3 above)
Comment (michaelkernaghan) [reply packages/taquito-contracts-library/test/taquito-contracts-library.spec.ts:108]: removed the line

File: packages/taquito-contracts-library/test/taquito-contracts-library.spec.ts (hunk 4 above)
Comment (michaelkernaghan) [reply packages/taquito-contracts-library/test/taquito-contracts-library.spec.ts:115]: added these expectations:

Comment (michaelkernaghan) [reply packages/taquito-contracts-library/test/taquito-contracts-library.spec.ts:115]: const contractData = contractLib.getContract(contractAddress);
    expect(contractData.entrypoints).toEqual(entrypoints);
    expect(contractData.script).toEqual(script);

File: packages/taquito-contracts-library/test/taquito-contracts-library.spec.ts (hunk 5 above)
Comment (michaelkernaghan) [reply packages/taquito-contracts-library/test/taquito-contracts-library.spec.ts:132]: This was from the previous test author.  I believe it means to check that there is no .code section of the script, t be sure it is invalid.  it says "expect the assertion that there is a script.code to be false"

File: packages/taquito-contracts-library/test/taquito-contracts-library.spec.ts (hunk 6 above)
Comment (michaelkernaghan) [reply packages/taquito-contracts-library/test/taquito-contracts-library.spec.ts:150]: This test adds nothing to code coverage and is mainly a repetition of an earlier test, except it adds the Tezos.addExtension,
I have removed the test.

File: packages/taquito-contracts-library/test/taquito-contracts-library.spec.ts (hunk 5 above)
Comment (roxaneletourneau) [reply packages/taquito-contracts-library/test/taquito-contracts-library.spec.ts:129]: Oh ok, as the line is green and I don't see it on master, I thought it was an addition in the current PR

**Review by roxaneletourneau — APPROVED** [review]

**Pull request merged**